New compose types

dashea opened the pull request:
This amalgamates https://github.com/weldr/lorax/pull/464, https://github.com/weldr/lorax/pull/463, and https://github.com/weldr/lorax/pull/466 so that they're not all mutually conflicting, removes the %post magic that didn't make sense from ami.ks, and uses that ami.ks (without the ec2-specific or bootloader parts) as the basis of vhd.ks and vmdk.ks.

larskarlitski commented:
Thanks! I can create ami images now, but get failures for vmdk and vhd.

For vmdk:
```
08:32:36,832 CRT exception: Traceback (most recent call last):

  File "/usr/lib/python3.6/site-packages/simpleline/render/screen_scheduler.py", line 245, in _process_screen
    top_screen.ui_screen.get_input_with_error_check(top_screen.args)

  File "/usr/lib/python3.6/site-packages/simpleline/render/screen/__init__.py", line 285, in get_input_with_error_check
    self._input_manager.get_input(args=args)

  File "/usr/lib/python3.6/site-packages/simpleline/render/screen/input_manager.py", line 91, in get_input
    prompt = self._ui_screen.prompt(args)

  File "/usr/lib64/python3.6/site-packages/pyanaconda/ui/tui/hubs/summary.py", line 119, in prompt
    raise CmdlineError(errtxt)

pyanaconda.errors.CmdlineError: The following mandatory spokes are not completed:
Installation source
Software selection
```

For vhd:
```
07:35:44,321 CRT exception: Traceback (most recent call last):

  File "/usr/lib64/python3.6/site-packages/pyanaconda/threading.py", line 286, in run
    threading.Thread.run(self)

  File "/usr/lib64/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)

  File "/usr/lib64/python3.6/site-packages/pyanaconda/installation.py", line 204, in doConfiguration
    configuration_queue.start()

  File "/usr/lib64/python3.6/site-packages/pyanaconda/installation_tasks.py", line 304, in start
    item.start()

  File "/usr/lib64/python3.6/site-packages/pyanaconda/installation_tasks.py", line 304, in start
    item.start()

  File "/usr/lib64/python3.6/site-packages/pyanaconda/installation_tasks.py", line 472, in start
    self.run_task()

  File "/usr/lib64/python3.6/site-packages/pyanaconda/installation_tasks.py", line 438, in run_task
    self._task(*self._task_args, **self._task_kwargs)

  File "/usr/lib64/python3.6/site-packages/pyanaconda/kickstart.py", line 2072, in execute
    util.enable_service(svc)

  File "/usr/lib64/python3.6/site-packages/pyanaconda/core/util.py", line 697, in enable_service
    raise ValueError("Error enabling service %s: %s" % (service, ret))

ValueError: Error enabling service cloud-init: 1
```

dashea commented:
Switched to just using partitioned-disk.ks for VHD and VMKD

atodorov commented:
@dashea if you rebase the glusterfs commit is not applied to master. Also there's a test which compares the list of supported image types which will probably fail.

dashea commented:
Rebased and updated the new compose types test for the new compose types.

dashea commented:
Pushed some new versions. Removed the commit that moves the AMI logic: the kind of AMI image we want is different from what --make-ami creates, so just bypass that option entirely and do ami composes as raw disk images. Modified the kickstarts to include more of what we understand now is needed. This removes the stuff that was copied from fedora spins' %post sections, some of that will probably need to be added back as we figure more out.

wgwoods commented:
Looks reasonable to me, although I do have one question: why are `cloud-init` and the virt guest agents being added to the generic qcow2 compose rather than in the specific kickstarts for AMI/VMDK/etc?

wgwoods commented:
..Microsoft specifically recommends removing `cloud-init` when building a VHD, but I don't know if that will _break_ anything? If it's easier to just merge this now and fix things like that as we go, that's fine with me.

stefwalter commented on share/composer/ami.ks:
```diff
@@ -0,0 +1,44 @@
+grub2
+
+chrony
+cockpit
```
Yes, lets remove Cockpit agree. This must have snuck in from the Fedora cloud kickstart.

stefwalter commented on share/composer/qcow2.ks:
```diff
@@ -37,4 +39,10 @@ kernel
 -dracut-config-rescue
 grub2
 
+cloud-init
```
Agree for qcow.ks ... but for ami.ks, vhd.ks I think we need to include cloud-init by default. I think the image wouldn't be properly accessible without it.

dashea commented:
Removed cockpit from ami and vhd, removed cloud-init from qcow2. For the /boot partition, I added /boot lines to ami.ks and vhd.ks and removed the logic from compose.py. I left the rootpw and auth lines as they were since the rootpw patch isn't currently on master; I'll remove those to match partitioned-disk.ks on rhel8-branch.
